app/src/main/java/org/fossasia/openevent/general/social/SocialLink.kt

```diff
@@ -1,22 +1,30 @@
 package org.fossasia.openevent.general.social
 
-import android.arch.persistence.room.*
+import android.arch.persistence.room.ColumnInfo
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.ForeignKey
+import android.arch.persistence.room.ForeignKey.CASCADE
+import android.arch.persistence.room.PrimaryKey
 import com.fasterxml.jackson.databind.PropertyNamingStrategy
 import com.fasterxml.jackson.databind.annotation.JsonNaming
 import com.github.jasminb.jsonapi.IntegerIdHandler
 import com.github.jasminb.jsonapi.annotations.Id
 import com.github.jasminb.jsonapi.annotations.Relationship
 import com.github.jasminb.jsonapi.annotations.Type
 import org.fossasia.openevent.general.event.Event
+import org.fossasia.openevent.general.event.EventId
 
 @Type("social-link")
 @JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
+@Entity
 data class SocialLink(
         @Id(IntegerIdHandler::class)
         @PrimaryKey
         val id: Int,
         val link: String,
         val name: String,
+        @ColumnInfo(index = true)
+        @ForeignKey(entity = Event::class, parentColumns = ["id"], childColumns = ["event"], onDelete = CASCADE, onUpdate = CASCADE)
```

Comment:
Not sure why this foreign key does not work, please revert back to your method of declaring foreign key

Reply:
Okay will do that :+1:  Was the error due to ColumnInfo ?

Reply:
No, JSONAPI converter can't set relationship at the time of creation of entity

Reply:
Ohhkay !! I verified that now :+1: So nullable made the difference